torchvision/datasets/folder.py

```diff
@@ -32,18 +32,24 @@ def is_image_file(filename):
     return has_file_allowed_extension(filename, IMG_EXTENSIONS)
 
 
-def make_dataset(dir, class_to_idx, extensions):
+def make_dataset(dir, class_to_idx, extensions=None, is_valid_file=None):
     images = []
     dir = os.path.expanduser(dir)
+    if extensions is None and is_valid_file is None:
+        raise ValueError("Both extensions and is_valid_file cannot be None")
```

Comment:
@fmassa can't we merge the two conditions into something like xor of both and showing the error message like "both extensions and is_valid_file cannot be active or inactive at the same time"

Reply:
Sure, we can improve on that part!